src/enumerable.cr

```diff
@@ -1099,7 +1099,7 @@ module Enumerable(T)
   end
 
   # Like `minmax` but returns `{nil, nil}` if the collection is empty.
-  def minmax? : {T?, T?}
+  def minmax? : {T, T} | {Nil, Nil}
```

Comment:
These are the same :-)

Reply:
That's non-trivial, since one might expect it to be `{T, T} | {T, Nil} | {Nil, T} | {Nil, Nil}` instead. This is still more readable.

Reply:
Yes, I think the focus should be on readbility as the return type restriction serves as documentation. THis clearly expresses that either both values are `Nil` or both are `T`.

Reply:
Okay, but the compiler will merge that union into `{T?, T?}` and there's no type-safe way to guarantee that both are nil or non-nil at the same time. 

A better type for that might have been `{T, T}?` but at this point it's a breaking change 😞

Reply:
Also, if a user reads that type and they actually print the `typeof` of it, they will see `{T?, T?}`. So I think this new way is actually more prone to confusion.

Reply:
very good point @asterite ! let's roll it back then